packages/ledger/src/v2.ts

```diff
@@ -0,0 +1,288 @@
+import {
+  Chain,
+  WalletInit,
+  EIP1193Provider,
+  ProviderAccounts,
+  createEIP1193Provider
+} from '@web3-onboard/common'
+import type {
+  EthereumProvider as LedgerEthereumProvider
+} from '@ledgerhq/connect-kit-loader'
+import { isHexString, LedgerOptionsWCv2 } from './index.js'
+import { JQueryStyleEventEmitter } from 'rxjs/internal/observable/fromEvent'
+
+// methods that require user interaction
+const defaultOptionalMethods = [
+  'eth_sendTransaction',
+  'eth_signTransaction',
+  'personal_sign',
+  'eth_sign',
+  'eth_signTypedData',
+  'eth_signTypedData_v4'
+]
+
+function ledger(options?: LedgerOptionsWCv2): WalletInit {
+  const projectId = options?.walletConnectVersion == 2 ? options.projectId : undefined
+  if (!projectId) {
+    throw new Error(
+      'WalletConnect requires a projectId. Please visit https://cloud.walletconnect.com to get one.'
+    )
+  }
+
+  return () => {
+    return {
+      label: 'Ledger',
+      getIcon: async () => (await import('./icon.js')).default,
+      getInterface: async ({ chains, EventEmitter }) => {
+        const {
+          loadConnectKit,
+          SupportedProviders,
+          SupportedProviderImplementations
+        } = await import('@ledgerhq/connect-kit-loader')
+
+        const connectKit = await loadConnectKit()
+        if (options?.enableDebugLogs) {
+          connectKit.enableDebugLogs()
+        }
+
+        // accept both hex and decimal chain ids
+        const requiredChains = options?.requiredChains?.map((id) => (
+          (typeof id === 'string' && isHexString(id))
+            ? parseInt(id, 16)
+            : id as number
+        ))
+
+        const checkSupportResult = connectKit.checkSupport({
+          providerType: SupportedProviders.Ethereum,
+          walletConnectVersion: 2,
+          projectId: options?.projectId,
+          chains: requiredChains,
+          optionalChains: chains.map(({ id }) => parseInt(id, 16)),
+          methods: options?.requiredMethods,
+          optionalMethods: options?.optionalMethods ||
+            defaultOptionalMethods,
+          events: options?.requiredEvents,
+          optionalEvents: options?.optionalEvents,
+          rpcMap: chains
+            .map(({ id, rpcUrl }) => ({ id, rpcUrl }))
+            .reduce((rpcMap: Record<number, string>, { id, rpcUrl }) => {
+              rpcMap[parseInt(id, 16)] = rpcUrl || ''
+              return rpcMap
+            }, {}),
+        })
+
+        // get the provider instance, it can be either the Ledger Extension
+        // or WalletConnect
+        const instance = (await connectKit.getProvider()) as LedgerEthereumProvider
+
+        // return the Ledger Extension provider
+        if (
+          checkSupportResult.providerImplementation ===
+            SupportedProviderImplementations.LedgerConnect
+        ) {
+          return {
+            provider: createEIP1193Provider(instance)
```

Comment:
Just checking that this is intended to be wrapped with the `createEIP1193Provider` method, since the ledger extension provider is not wrapped in the [v1 file here](https://github.com/blocknative/web3-onboard/pull/1549/files#diff-e410806c59fd896721912f63f406b86661d71a193ed324b6f6f250e6556791e7R48)?

Reply:
Well spotted, updated and tested it.